theano/gpuarray/blas.py

```diff
@@ -52,6 +53,15 @@ def make_node(self, y, alpha, A, x, beta):
         y = as_gpuarray_variable(y, ctx_name)
         alpha = as_tensor_variable(alpha).astype('float64')
         beta = as_tensor_variable(beta).astype('float64')
+
+        # if alpha==1. and beta==0., we add runtime check
+        # for possible speed up using vector-vector dot as
+        # gemv tend to be slower for vector-vector dot
+        self._use_rt_dot_check = False
+        if all(map(lambda v: isinstance(v, TensorConstant), [alpha, beta])):
+            if alpha.value == 1. and beta.value == 0.:
+                self._use_rt_dot_check = True
+
```

Comment:
why this? I would remove this check and always do the case this is True. This way, the speed up will be there even if we don't know the shapes at compile time.